core/src/main/java/org/mule/DefaultMuleMessage.java

```diff
@@ -1322,6 +1353,24 @@ private void writeObject(ObjectOutputStream out) throws Exception
         return toWrite;
     }
 
+    @Override
+    protected Serializable serializeValue(ObjectOutputStream out) throws Exception
+    {
+        byte[] serializablePayload = muleContext.getTransformationService().getPayloadAsBytes(this);
```

Comment:
where is this used? I couldn't find usages of this method in the PR. Other than that, the method name doesn't seem correct to me. This logic should only be performed when the payload is not serializable.